src/django/pfb_analysis/models.py

```diff
@@ -93,10 +95,43 @@ class Status(object):
     neighborhood = models.ForeignKey(Neighborhood,
                                      related_name='analysis_jobs',
                                      on_delete=models.CASCADE)
-    status = models.CharField(choices=Status.CHOICES,
-                              default=Status.CREATED,
-                              max_length=12,
-                              help_text='The current status of the AnalysisJob')
+
+    @property
+    def status(self):
+        """ Return current status for this job
+
+        Combines the sources of status information (updates, batch job status, inference) to
+        return the best answer.
+        """
+        # If there's no job ID that means it's brand new
+        if self.batch_job_id is None:
+            return self.Status.CREATED
+
+        if self.batch_job_status == 'FAILED':
+            return self.Status.ERROR
+        try:
+            # If the job hasn't failed and has sent any status updates, use the latest
+            return self.status_updates.last().status
+        except AttributeError:
+            # Not failed but no status updates means it must be in the queue
+            return self.batch_job_status
```

Comment:
Do we want to be mixing AnalysisJob status and batch job status here? Would it make more sense to keep those two things completely separate, and just add an additional `status` that encapsulates the job being queued?